### DISCUSSION TITLE
How to prevent submit enhanced form when hit enter in an input field?

### DISCUSSION THREAD

**Discussion by NormandoHall** [sveltejs/kit#8657] [General]
I like enhanced form SK provides, but it is impossible to prevent submit (with cancel() in this case) when the user hit ENTER in any field. I want the user only submit the form when click the submit button. 

Based on [this answer in Stack Overflow](https://stackoverflow.com/a/51507806/5348357), I workaround adding a disabled button at the beginning of the form:

`<button type="submit" disabled style="display: none" />`

But looks very weird. Is there a better way to get the same behaviour?

Regards

**Comment by teemingc** [comment]
```
<button type="submit" disabled style="display: none" aria-hidden="true"></button
```
I like the answer you found that includes the aria attribute and works without JavaScript.

However, you can also implement the other answer with svelte if you prefer, such as
```svelte
<!-- Disallow enter key anywhere -->
<form on:keydown={(event) => event.key != 'Enter'}>

</form>
```

**Reply by NormandoHall** [reply]
@s3812497 superb for the keydown event!!! Thanks!
